returnn/datasets/distrib_files.py

```diff
@@ -140,6 +151,13 @@ def __init__(
         :param get_sub_epoch_dataset: callable which returns a dataset dict for a given subset of files
         :param preload_next_n_sub_epochs: how many sub epoch datasets to preload
         :param buffer_size: buffer size for each worker, amount of seqs to prefetch
+        :param num_shards: shard the data into this many pieces and distribute between worker processes
```

Comment:
I don't think it makes sense to specify the num shards? You either enable sharding or not, and index is always based on rank of the distributed worker, and num workers is always the size.

Reply:
Yeah I was debating this. I think it might be a valid use case to have e.g. 16 workers, and only shard the data into 4 or 8 parts if this makes it realistic that every one worker can complete a full epoch. Not sure though, and removing the `int` option would simplify some of the validation logic.

Reply:
But this is effectively then just running multiple epochs? I don't really understand the use case.